server/service/store/handler_store_data.go

```diff
@@ -0,0 +1,103 @@
+package store
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func HandleStoreData(w http.ResponseWriter, r *http.Request) {
+	dataHashStr := mux.Vars(r)["key"]
+	_, err := hex.DecodeString(dataHashStr)
+	if err != nil {
+		log.Errorf("failed to decode dataHash: %s", err.Error())
+		http.Error(w, "failed to decode dataHash", http.StatusBadRequest)
+		return
+	}
+
+	dealIDStr := mux.Vars(r)["dealId"]
+	_, err = strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		log.Errorf("failed to parse deal ID: %s", err.Error())
+		http.Error(w, "failed to parse deal ID", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
```

Comment:
do not read all. pipe it

Reply:
Danke. The piping is also called as streaming. You may be able to google how to pipe io.Reader with io.Writer.

Reply:
I just saw what you did in the code comment. Yes, that's right to use io.Copy.

Reply:
Thanks I got it. I applied in [45f3e8f](https://github.com/medibloc/panacea-dep-consumer/pull/1/commits/45f3e8fee4349ef84e082bf41b65012ed465abc0).